chronicle/Sensor/SensorDataManager.swift

```diff
@@ -0,0 +1,225 @@
+//
+//  SensorDataManager.swift
+//  SensorDataManager
+//
+//  Created by Alfonce Nzioka on 11/10/21.
+//
+
+//
+//  SensorDataManager.swift
+//  SensorDataManager
+//
+//  Created by Alfonce Nzioka on 11/10/21.
+//
+
+import Foundation
+import SensorKit
+import OSLog
+import CoreData
+
+// handles all functionality related to sensor data: user authorization, configuring sensors for recording, fetching recorded data to Core Data, and uploading to server based on scheduled timer
+// TODO: configure authorized sensors to record data: https://developer.apple.com/documentation/sensorkit/srsensorreader
+
+class SensorDataManager: NSObject, SRSensorReaderDelegate {
+    
+    // access to CoreData stack
+    var persistenceController: PersistenceController = .shared
+    
+    // shared instance
+    static var shared = SensorDataManager()
+    
+    // due to CoreData concurrency issues, we want to prevent multiple threads from attempting to fetch the same data before the server's upload request returns
+    private var uploadingData: Bool
+    
+    private init(uploadingData: Bool = false) {
+        self.uploadingData = uploadingData
+    }
+    
+    // TODO: replace this with values from EDM
+    // https://github.com/openlattice/chronicle-ios/pull/6 fetches required PTIDs and saves them on device
+    let variablePTID = "f3c1a844-8c32-4d7a-b18f-68d4cc031daf"
+    let valuesPTID = "dcc3bc24-3a5d-45cf-8e38-bc9ba8c43d06"
+    let endDateTimePTID = "0ee3acba-51a7-4f8d-921f-e23d75b07f65"
+    let startDateTime = "92a6a5c5-b4f1-40ce-ace9-be232acdce2a"
+    let idPTID = "39e13db7-a730-421a-a600-ae0674060140"
+    let namePTID = "ddb5d841-4c82-407c-8fcb-58f04ffc20fe"
+    let dateLoggedPTID = "e90a306c-ee37-4cd1-8a0e-71ad5a180340"
```

Comment:
Don't these already exist in the `PropertyTypeIds` struct in `Mocks.swift`? Can you use those here?

Reply:
Deleting this since it's not used